impl/src/planning/subplanners.rs

```diff
@@ -446,182 +437,163 @@ impl<'planner> CrateSubplanner<'planner> {
         continue;
       }
 
-      // UNWRAP: Guaranteed to exist by checks in WorkspaceSubplanner#produce_build_plan
-      let buildable_target = self
-        .crate_catalog
-        .entry_for_package_id(dep_id)
-        .unwrap()
-        .workspace_path_and_default_target(&self.settings)?;
-
-      // Implicitly dependencies are on the [lib] target from Cargo.toml (of which there is
-      // guaranteed to be at most one).
-      // In this function, we don't explicitly narrow to be considering only the [lib] Target - we
-      // rely on the fact that only one [lib] is allowed in a Package, and so treat the Package
-      // synonymously with the [lib] Target therein.
-      // Only the [lib] target is allowed to be labelled as a proc-macro, so checking if "any"
-      // target is a proc-macro is equivalent to checking if the [lib] target is a proc-macro (and
-      // accordingly, whether we need to treat this dep like a proc-macro).
-      let is_proc_macro = dep_package
-        .targets
-        .iter()
-        .flat_map(|target| target.crate_types.iter())
-        .any(|crate_type| crate_type.as_str() == "proc-macro");
-
-      let buildable_dependency = BuildableDependency {
-        name: dep_package.name.clone(),
-        version: dep_package.version.clone(),
-        buildable_target: buildable_target.clone(),
-        is_proc_macro,
-      };
-
-      if build_dep_names.contains(&dep_package.name) {
-        if buildable_dependency.is_proc_macro {
-          dep_set
-            .build_proc_macro_deps
-            .push(buildable_dependency.clone());
-        } else {
-          dep_set.build_deps.push(buildable_dependency.clone());
+      // TODO(GregBowyer): Reimplement what cargo does to detect bad renames
```

Comment:
The problem manifests as this

```toml
[dependencies]
bytes_new = { version = "0.3.0", package = "bytes" }
bytes_old = { version = "0.3.0", package = "bytes" }
```

Strictly this is an error. Right now cargo metadata will basically lose both `bytes` deps from the resolve graph :gruntle: but a `cargo build` will scream about having the same basic package with multiple names.

I dont have a good solution right now (I dont think we should go back to trying to essentially emulate the resolve graph like before), so I guess a todo :/

Reply:
+1 to TODO(#cargo-raze-issue): referencing this error.